bus-mapping/src/rpc.rs

```diff
@@ -152,53 +189,71 @@ mod rpc_tests {
     // have a Geth client synced with mainnet to have unified "test-vectors".
     #[ignore]
     #[tokio::test]
-    async fn test_get_block_by_hash() {
-        let transport = Http::new(Url::parse("http://localhost:8545").unwrap());
-
-        let hash = Hash::from_str("0xe4f7aa19a76fcf31a6adff3b400300849e39dd84076765fb3af09d05ee9d787a").unwrap();
-        let prov = GethClient::new(transport);
-        let block_by_hash = prov.get_block_by_hash(hash).await.unwrap();
-        assert!(hash == block_by_hash.hash.unwrap());
+    async fn test_get_block_by_number() {
+        let prov = get_provider();
+        let block_by_num_latest =
+            prov.get_block_by_number(BlockNumber::Latest).await.unwrap();
+        let block_by_num = prov.get_block_by_number(2u64.into()).await.unwrap();
+        assert!(
+            block_by_num.transactions[0].hash
+                == block_by_num_latest.transactions[0].hash
+        );
     }
 
     // The test is ignored as the values used depend on the Geth instance used
     // each time you run the tests. And we can't assume that everyone will
     // have a Geth client synced with mainnet to have unified "test-vectors".
     #[ignore]
     #[tokio::test]
-    async fn test_get_block_by_number() {
-        let transport = Http::new(Url::parse("http://localhost:8545").unwrap());
-
-        let hash = Hash::from_str("0xe4f7aa19a76fcf31a6adff3b400300849e39dd84076765fb3af09d05ee9d787a").unwrap();
-        let prov = GethClient::new(transport);
+    async fn test_get_block_by_hash() {
+        let prov = get_provider();
         let block_by_num_latest =
             prov.get_block_by_number(BlockNumber::Latest).await.unwrap();
-        assert!(hash == block_by_num_latest.hash.unwrap());
-        let block_by_num = prov.get_block_by_number(1u64.into()).await.unwrap();
-        assert!(
-            block_by_num.transactions[0].hash
-                == block_by_num_latest.transactions[0].hash
-        );
+        let block_by_hash = prov
+            .get_block_by_hash(block_by_num_latest.hash.unwrap())
+            .await
+            .unwrap();
```

Comment:
That's exactly what you should be doing in the `test_get_block_by_number` instead of hardcoding the `2`.

You can use the same strategy! :+1: